### DISCUSSION TITLE
Query Result column in Database

### DISCUSSION THREAD

**Discussion by jrneal1** [celery/celery#7568] [Q&A] [answered]
I'm working on storing my results in a SQL Server database.  the result column is a VARBINARY(MAX) datatype.  Is it possible to write a SQL Query to convert this data to readable text?

I know how to use AsyncResult to get it in python.  However, I want to be able to do get the result outside of python.

An example result = 0x80049509000000000000008C056569646153942E

Is this JSON?  Is it base64 encoded?  UTF-8?  I notice all the entries start with 0x800495.  Googling that doesn't yield any results.  Just having a hard time knowing where to start.  Just need someone to point me in the right direction.

UPDATE: The hex example above is pickled data.  

[https://stackoverflow.com/questions/55533906/unpickle-element-that-was-pickled-with-sqlalchemy-by-hand](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fstackoverflow.com%2Fquestions%2F55533906%2Funpickle-element-that-was-pickled-with-sqlalchemy-by-hand&data=05%7C01%7Cjneal%40tva.gov%7C44c60539654e4a8d8a7508da4e1e411e%7C270992cd9003497184ded1640c0bffc5%7C0%7C0%7C637908189223116289%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000%7C%7C%7C&sdata=rF1cyjh%2FoB%2BcfXvISNo6tA%2BuUN05VLbLmz7uAe6Ey9o%3D&reserved=0)

```python
my_string = '0x80049509000000000000008C056569646153942E'

pickled_data = bytes.fromhex(my_string[2:])

obj = pickle.loads(pickled_data)
print(obj)
```

So, new questions - 1) Anyone know the TSQL equivalent to the this python code?  2) I know this is celery and not sqlalchemy, but does anyone know if there's a way to prevent sqlalchemy from pickling the result or use json instead of pickle before writing it to the database?

**Comment by k-sparrow** [comment, marked as answer]
I faced the same problem using Postgres. It's actually coming from [SQLAlchemy](https://github.com/celery/celery/blob/main/celery/backends/database/models.py#L24).

There is a [long going discussion](https://github.com/celery/celery/pull/5907) on replacing the `result` column type from `PickleType` to a plain binary field and have the serialization done on Celery's side using the configured result serializer, but it hasn't resolved yet.

In the meantime, I personally ended up patching the `DatabaseBackend` (`PickleType` -> `Text`) locally and plugged it as a custom backend.
